### PR TITLE
Quote post API

### LINKED ISSUE #207: Add quote posting API
### MOTIVATION DIVISION.

We've discussed the integration questions with @leha-bot and have agreed on the following implementation to integrate loglist with [@xthon channel](https://t.me/xthon) in Telegram.

### IMPLEMENTATION DIVISION.

1. Add a new API to post the quote to server. Posted quote should be initially unapproved. API should require secret token to be posted.
2. Add a new field to the quotes: `source`. The field should be automatically filled when posting the quote. Pairs "API Token → Source" should be kept in the database.
3. When returning any quotes from the external API, add the `source` field to them.
4. Tag the quotes with source on the UI.

### END PROGRAM.

### PR DESCRIPTION
**Pull request by ForNeVeR**
Closes #207.

The main page now looks like this:

image

### CODE REVIEW

File: scalajvm/app/services/SuggestedQuoteService.scala [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+object SuggestedQuoteService {
+
+  def insertAndNotify(text: String, remoteAddress: String, source: String)
+                     (implicit request: RequestHeader, session: DBSession): Unit = {
+    val id = SuggestedQuoteQueries().insertSuggestedQuote(text, remoteAddress, source)
+    SuggestedQuoteQueries().getSuggestedQuoteById(id) match {
+      case Some(suggestedQuote) => {
+        val approvers = ApproverQueries().getAllApprovers
+        Notifications.notifyApproversAboutSuggestedQuote(approvers, suggestedQuote)
```
Comment (rexim) [inline scalajvm/app/services/SuggestedQuoteService.scala]: How about to include the source to the notification email? So the admins know where it came from. :)

**Comment by rexim** [comment]
After a brief look I'm ok with the changes, but I wanna take a closer a little bit later. Thanks!

File: docs/API.md [hunk 2]
```diff
@@ -42,6 +42,28 @@ API Methods
 
 ### Quote access
 
+#### Post the quote
+
+Send `POST` request to `/api/quote/new`. Quote should have the following form:
+
+```json
+{
+    "text": "quote test",
```
Comment (Minoru) [inline docs/API.md]: This should say "quote te**x**t", shouldn't it?

Comment (ForNeVeR) [inline docs/API.md]: Well, that wasn't intended, but probably yes. I'll fix that.

**Comment by ForNeVeR** [comment]
For some reason Travis refuses to provide status for our builds. I'll reinitialize Travis in this repository.

**Comment by rexim** [comment]
@ForNeVeR how API key is generated?

**Comment by ForNeVeR** [comment]
API key should be generated manually and put into the `api_key` table manually.

- we couldn't do it with migration because API key should be private
- any user-accessible UI and/or API for that wasn't part of the task

Probably it's a good idea to make new (basic auth-protected) API to insert the key/source pair into our database.

**Comment by rexim** [comment]
> Probably it's a good idea to make new (basic auth-protected) API to insert the key/source pair into our database.

We can have a separate issue for that

**Comment by ForNeVeR** [comment]
I agree; I'll make the necessary changes tomorrow. Thanks!

**Comment by ForNeVeR** [comment]
It has been done.

I'm merging that, but feel free to leave any additional suggestions if you think something else is missing.

**Comment by ForNeVeR** [comment]
Thanks for the review, everyone! \o/

**Pull request merged**